tests: Context Gateway diff tests fail on windows-latest (CRLF + path-separator)

## What

The `test (windows-latest)` CI job fails on `main` in `packages/memtomem/tests/test_web_routes_context.py` — 4 Context Gateway diff tests break on Windows-specific newline and path-separator handling. The job is `continue-on-error` and not in the required-check set, so the red is silent: it shows up on every PR as a carry-over failure (e.g. it was initially mistaken for a PR-side failure while reviewing #1251, which is docs-only).

Example run (main): https://github.com/memtomem/memtomem/actions/runs/27329712025

## Failing tests

```
FAILED test_web_routes_context.py::TestDiffCommand::test_diff_project_local_has_no_runtime_rows[asyncio]
  assert '---\r\ndescr...ARGUMENTS\r\n' == '---\ndescrip...n$ARGUMENTS\n'
FAILED test_web_routes_context.py::TestDiffAgent::test_malformed_canonical_reports_parse_error_with_reason[asyncio]
  AssertionError: assert '.memtomem\\agents\\broken.md' == '.memtomem/agents/broken.md'
FAILED test_web_routes_context.py::TestDiffAgent::test_healthy_canonical_payload_has_canonical_path_and_no_reason[asyncio]
  AssertionError: assert '.memtomem\\agents\\fine.md' == '.memtomem/agents/fine.md'
FAILED test_web_routes_context.py::TestDiffAgent::test_diff_project_local_has_no_runtime_rows[asyncio]
  AssertionError: assert '---\nname:...oroughly.\r\n' == '---\nname: r...thoroughly.\n'
```

Two distinct shapes:

1. **CRLF in diff content** — the content the diff endpoint returns carries `\r\n` on Windows while the expected strings use `\n`.
2. **Backslash in reported paths** — the payload's canonical path is `os.sep`-joined (`.memtomem\agents\...`) while the assertions expect POSIX `/`.

## Open question (needs a decision before fixing)

It's not yet established which side is wrong — both hypotheses below need verification against the source:

- If the **web payload contract** is "POSIX separators + LF" (which the web UI consuming these fields may already assume), the fix belongs in the route/diff code (normalize path display and read canonical files with `newline=""` or normalize to `\n`), and the tests are correctly pinning the contract.
- If the payload is intentionally platform-native, the tests should normalize before comparing (and the web UI's handling of `\\` paths and `\r\n` content should be checked).

Note the CRLF may originate in the test fixtures themselves (text-mode writes translate `\n` → `\r\n` on Windows), in which case fixture writes need `newline="\n"` regardless of the contract decision.

## Done when

- The 4 tests pass on `windows-latest` without weakening what they pin on POSIX.
- The chosen contract (platform-native vs normalized) is stated in the diff payload code or test module docstring.

--- a/packages/memtomem/src/memtomem/web/routes/context_agents.py
+++ b/packages/memtomem/src/memtomem/web/routes/context_agents.py
@@ -537,6 +537,13 @@ async def update_agent(
 
 
 def _safe_rel(p: Path, project_root: Path) -> str:
+    """Project-relative path as a POSIX string for API payloads.
+
+    ``.as_posix()`` (not ``str``) so canonical/runtime paths come back
+    ``/``-separated on every platform — the Web UI and diff payloads pin POSIX
+    separators (#1256). Falls back to the absolute POSIX path for user-tier
+    locations outside ``project_root``.
+    """
     try:
         return p.relative_to(project_root).as_posix()
     except ValueError:

--- a/packages/memtomem/src/memtomem/web/routes/context_commands.py
+++ b/packages/memtomem/src/memtomem/web/routes/context_commands.py
@@ -100,6 +100,13 @@ def _reject_project_local_write(target_scope: TargetScope, action: str) -> None:
 
 
 def _safe_rel(p: Path, project_root: Path) -> str:
+    """Project-relative path as a POSIX string for API payloads.
+
+    ``.as_posix()`` (not ``str``) so canonical/runtime paths come back
+    ``/``-separated on every platform — the Web UI and diff payloads pin POSIX
+    separators (#1256). Falls back to the absolute POSIX path for user-tier
+    locations outside ``project_root``.
+    """
     try:
         return p.relative_to(project_root).as_posix()
     except ValueError:

--- a/packages/memtomem/tests/test_web_routes_context.py
+++ b/packages/memtomem/tests/test_web_routes_context.py
@@ -1,4 +1,18 @@
-"""Tests for Context Gateway web routes (overview + skills + commands + agents)."""
+"""Tests for Context Gateway web routes (overview + skills + commands + agents).
+
+Diff-payload contract pinned by ``TestDiffCommand`` / ``TestDiffAgent`` (#1256):
+
+- **Paths** are normalized to POSIX separators on every platform. The route
+  helpers ``_safe_rel`` in ``context_commands`` / ``context_agents`` return
+  ``.as_posix()`` so the Web UI receives stable ``/``-separated paths; the diff
+  tests assert literal POSIX strings (e.g. ``.claude/commands/ghost.md``) rather
+  than ``str(Path(...))``, which would be backslash-joined on Windows.
+- **Content** is LF, not platform-native. Canonical generators and sync write
+  LF bytes, so fixtures here use ``_write_text_lf`` (raw bytes) instead of
+  ``Path.write_text`` (text mode translates ``\\n`` -> ``\\r\\n`` on Windows).
+  The original Windows CRLF breakage was that fixture artifact, not the route
+  re-normalizing newlines.
+"""
 
 from __future__ import annotations
 

